docs/adrs/00007-purl-report.md

```diff
@@ -0,0 +1,104 @@
+# 00007. PURL vulnerability report
+
+Date: 2025-05-21
+
+## Status
+
+PROPOSED
+
+## Context
+
+The UI should have a feature for users to "upload" their SBOM, and get a report about vulnerabilities of contained
+components. The term "upload" may be misleading, as the backend must never store the document.
+
+"Dependency Analytics" requires a way to check a set of PURLs for their vulnerabilities too. Ideally, we can somehow
+re-use that functionality.
+
+As processing the correlation information may require some time, we may encounter request timeouts, like we sometimes
+see for larger SBOMs. The idea is to leverage the same pattern as we use for the UI upload. Upload with a `watch=true`
+flag, and then poll another API for the outcome.
+
+### Assumptions
+
+This ADR makes the following assumptions:
+
+* The provided SBOM will only be scanned for PURLs, not for CPEs, hashes/digests or other references
+
+## Decision
+
+There's a two-step process. First, upload (but don't store) the SBOM to the backend. The backend will parse the
+document and extract all PURLs, returning them to the client. This process, depending on the size of the file, is
+rather fast. The request shouldn't time out, as either the document or the response is being transferred. The actual
+processing, without any transfer, should be quick.
+
+The client can also list the result, once returned, to the user. To show the user what was understood from the SBOM.
+But that's an optional approach.
+
+Next, the client will open a websocket connection to the backend. It can then send as many PURLs as it wants to the 
```

Comment:
Another solution is to send requests in batches, this is how other vulnerability providers are implemented (examples are OSS Sonatype or OSV, if I recall correctly). Maybe batch requests will be more efficient than 1 by 1 queries.

DA Exhort can easily adapt the current implementation into batch requests as it has already been done for the other providers.

Reply:
Ah yes. That's already what I changed it to. But didn't seem to push that change. I'd prefer that too.